### PR TITLE
Remove unnecessary ProjectReferences to CoreCLRTestLibrary

### PR DESCRIPTION
**Pull request by echesakovMSFT**
Remove the **ProjectReference**-s to CoreCLRTestLibrary.csproj in those project files where the CoreCLRTestLibrary is not actually being used.

Each **ProjectReference** item is ended up in **PrepareProjectReferences** target that creates unnecessary items.

### CODE REVIEW

File: tests/src/baseservices/typeequivalence/simple/Simple.cs [hunk 1]
```diff
@@ -8,7 +8,6 @@
 using System.Reflection;
 using System.Runtime.InteropServices;
 
-using TestLibrary;
```
Comment (AaronRobinsonMSFT) [inline tests/src/baseservices/typeequivalence/simple/Simple.cs]: I am pretty sure this is needed for the `Assert` call below - at least that is what I intended when I wrote this test.

Comment (echesakovMSFT) [reply tests/src/baseservices/typeequivalence/simple/Simple.cs]: Ah ok. Then I need to revert changes for this project. 

I tested all the changes on Linux machine and they succeeded. However, this test is built as Windows only, so I didn't catch the error earlier.

Comment (AaronRobinsonMSFT) [reply tests/src/baseservices/typeequivalence/simple/Simple.cs]: >I tested all the changes on Linux machine and they succeeded

Oops. Yeah all COM and TypeEquivalence tests are Windows only. There are also some other interop tests that have some tricky behavior on non-Windows - just FYI.

**Pull request merged**